sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/DebugAssertHasNoSideEffects.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Test()
+        {
+            Debug.Assert2(true);
+            Debug.Assert(list.Contains("a"));
+            Debug.Assert(list.Contains("a"));
+            Debug.Assert(list.DoStuff());
+            Debug.Assert(list.Destroy());
+
+            Debug.Assert(list.Remove("a")); // Noncompliant {{Expressions used in 'Debug.Assert' should not produce side effects.}}
+//                      ^^^^^^^^^^^^^^^^^^
```

Comment:
That's not a big deal, but perhaps it is better to underline only the argument of Debug.Assert(), without the parenthesis, e.g. one chevron to the right and two chevrons shorter.